src/app/record/array-record/array-record.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { JsonPatch } from '../../shared/interfaces';
+import { BaseRecord } from '../base-record';
+
+@Component({
+  selector: 'array-record',
+  templateUrl: './array-record.component.html',
+  styleUrls: ['./array-record.component.scss','../common-styles.scss']
+})
+export class ArrayRecordComponent extends BaseRecord implements OnInit {
+  @Input() record: any;
+  @Input() diff?: JsonPatch[];
+  @Input() depth: number;
+  @Input() expanded: boolean;
+  keys: Array<string> = [];
+  isEdited: boolean;
+
+  ngOnInit() {
+    if (this.diff) {
+      this.diff.forEach(item => {
+        if ( item.path.length <= this.depth){
+          this.isEdited = true;
+        }
+        else if (item.path.length > this.depth) {
+          this.diffKeys.set(item.path[this.depth],true);
+        }
+        if (item.path.length === (this.depth + 1) && item.op === 'add') {
+            this.record.push(item.value);
+        }
+        this.diffType = item.op;
```

Comment:
I disagree setting this class property populating in a for loop each time getting overwritten.